server/routerlicious/packages/services-shared/src/startupChecker.ts

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export class StartupChecker {
+	private static instance: StartupChecker;
+	private isReady: boolean = false;
+
+	private constructor() {}
+
+	public static getInstance(): StartupChecker {
```

Comment:
I had implemented this as a singleton so that we need not pass the object instance to every service as a resource. This way each service need not a new resource for the startupChecker. Instead, the startup endpoint can leverage this singleton instance in `healthCheckEndpoint` without having it being passed as a new resource.